Pull request for Charlie Keaney's branch

CharKeaney opened the pull request:
Author: Charlie Keaney <charlie.keaney@embecosm.com>

    Made Zca and C being equivalent instruction classes explicit.
    
        * include/opcode/riscv.h: Made Zca and C being equivalent instruction classes explicit.
        * opcodes/riscv-opc.c: Made Zca and C being equivalent instruction classes explicit.

    Fixed typo in zcmb/zcmpe error message.
    
        * bfd/elfxx-riscv.c: Fixed typo in error message.
            * gas/testsuite/gas/riscv/zc-zcmb-fail-arch-conflict-0.l: Fixed typo in error message.
            * gas/testsuite/gas/riscv/zc-zcmb-fail-arch-conflict-1.l: Fixed typo in error message.
            * gas/testsuite/gas/riscv/zc-zcmpe-fail-arch-conflict-1.l: Fixed typo in error message.
          
    Added test for successful assembly of cm.jt and cm.jalt instructions.
    
            * gas/riscv/zc-zcmt-jalt.d: Added test for successful assembly of instruction.
            * gas/riscv/zc-zcmt-jalt.s: Added test for successful assembly of instruction.
            * gas/riscv/zc-zcmt-jt.d: Added test for successful assembly of instruction.
            * gas/riscv/zc-zcmt-jt.s: Added test for successful assembly of instruction.
            * 
    Added tests for failure on invalid immediate offset outside of range for Zc* extensions.
    
            * gas/riscv/zc-zcb-c-lbu-range-fail.d: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-c-lbu-range-fail.l: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-c-lbu-range-fail.s: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-c-lh-range-fail.d: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-c-lh-range-fail.l: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-c-lh-range-fail.s: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-c-lhu-range-fail.d: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-c-lhu-range-fail.l: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-c-lhu-range-fail.s: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-c-sb-range-fail.d: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-c-sb-range-fail.l: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-c-sb-range-fail.s: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-c-sh-range-fail.d: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-c-sh-range-fail.l: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-c-sh-range-fail.s: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-cm-lh-range-fail.d: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-cm-lh-range-fail.l: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-cm-lh-range-fail.s: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-cm-lhu-range-fail.d: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-cm-lhu-range-fail.l: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-cm-lhu-range-fail.s: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcb-cm-lu-range-fail.l: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcmt-jalt-range-fail.d: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcmt-jalt-range-fail.l: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcmt-jalt-range-fail.s: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcmt-jt-range-fail.d: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcmt-jt-range-fail.l: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcmt-jt-range-fail.s: Added test for failure on invalid immediate offset outside of range.
            * gas/riscv/zc-zcmt-march-accept.d: Added test case for success on use of valid opcodes for Zcmt.
            * gas/riscv/zc-zcmt-march-accept.s: Added test case for success on use of valid opcodes for Zcmt.
            * gas/riscv/zc-zcmt-march-reject.d: Added test case for failure on use of invalid opcodes for Zcmt.
            * gas/riscv/zc-zcmt-march-reject.l: Added test case for failure on use of invalid opcodes for Zcmt.
            * gas/riscv/zc-zcmt-march-reject.s: Added test case for failure on use of invalid opcodes for Zcmt.

    Fixed bug that permitted use of invalid operands for cm.jt
    
            * gas/config/tc-riscv.c: Fixed bug that allowed use of invalid operand 64 with cm.jt.

    Added test case for failure/success on use of invalid/valid opcodes respectively for Zcmt.
    
        * gas/testsuite/gas/riscv/zc-zcmt-march-accept.d: Added test case for success on use of valid opcodes for Zcmt.
            * gas/testsuite/gas/riscv/zc-zcmt-march-accept.s: Added test case for success on use of valid opcodes for Zcmt.
            * gas/testsuite/gas/riscv/zc-zcmt-march-reject.d: Added test case for failure on use of invalid opcodes for Zcmt.
            * gas/testsuite/gas/riscv/zc-zcmt-march-reject.l: Added test case for failure on use of invalid opcodes for Zcmt.
            * gas/testsuite/gas/riscv/zc-zcmt-march-reject.s: Added test case for failure on use of invalid opcodes for Zcmt.

    Added test case for failure/success on use of invalid/valid opcodes respectively for Zcmpe.
    
            * gas/riscv/zc-zcmpe-march-accept.d: Added test case for success on use of valid opcodes for Zcmpe.
            * gas/riscv/zc-zcmpe-march-accept.s: Added test case for success on use of valid opcodes for Zcmpe.
            * gas/riscv/zc-zcmpe-march-reject.d: Added test case for failure on use of invalid opcodes for Zcmpe.
            * gas/riscv/zc-zcmpe-march-reject.l: Added test case for failure on use of invalid opcodes for Zcmpe.
            * gas/riscv/zc-zcmpe-march-reject.s: Added test case for failure on use of invalid opcodes for Zcmpe.
           
    Added test case for failure/success on use of valid/invalid opcodes for Zcmp respectively.
    
            * gas/riscv/zc-zcmp-march-accept.d: Added test case for success on use of valid opcodes for Zcmp.
            * gas/riscv/zc-zcmp-march-accept.s: Added test case for success on use of valid opcodes for Zcmp.
            * gas/riscv/zc-zcmp-march-reject.d: Added test case for failure on use of invalid opcodes for Zcmp.
            * gas/riscv/zc-zcmp-march-reject.l: Added test case for failure on use of invalid opcodes for Zcmp.
            * gas/riscv/zc-zcmp-march-reject.s: Added test case for failure on use of invalid opcodes for Zcmp.
            
    Added test case for failure/success on use of invalid/valid opcodes respectively for Zcmb.
    
            * gas/riscv/zc-zcmb-march-accept.d: Added test case for success on use of valid opcodes for Zcmb.
            * gas/riscv/zc-zcmb-march-accept.s: Added test case for success on use of valid opcodes for Zcmb.
            * gas/riscv/zc-zcmb-march-reject.d: Added test case for failure on use of invalid opcodes for Zcmb.
            * gas/riscv/zc-zcmb-march-reject.l: Added test case for failure on use of invalid opcodes for Zcmb.
            * gas/riscv/zc-zcmb-march-reject.s: Added test case for failure on use of invalid opcodes for Zcmb.

    Added test cases for success on use of valid opcodes for Zcf.
    
        * gas/riscv/zc-zcf-march-accept.d: Added test case for success on use of valid opcodes for Zcf.
            * gas/riscv/zc-zcf-march-accept.s: Added test case for success on use of valid opcodes for Zcf.
            * gas/riscv/zc-zcf-march-reject.d: Added test case for failure on use of invalid opcodes for Zcf.
            * gas/riscv/zc-zcf-march-reject.l: Added test case for failure on use of invalid opcodes for Zcf.
            * gas/riscv/zc-zcf-march-reject.s: Added test case for failure on use of invalid opcodes for Zcf.
            
    Added test case for failure or success on use of invalid or valid opcodes respectively for Zcb.

    Added test cases for invalid opcodes on Zca
    
            * gas/riscv/zc-zca-c-flw-c-flwsp-c-fsw-c-fswsp-fail-march.d: Added test case for failure on use of invalid opcodes for Zca.
            * gas/riscv/zc-zca-c-flw-c-flwsp-c-fsw-c-fswsp-fail-march.l: Likewise.
            * gas/riscv/zc-zca-c-flw-c-flwsp-c-fsw-c-fswsp-fail-march.s: Likewise.
            * gas/riscv/zc-zca-c-mul-fail-march.d: Likewise.
            * gas/riscv/zc-zca-c-mul-fail-march.l: Likewise.
            * gas/riscv/zc-zca-c-mul-fail-march.s: Likewise.
            * gas/riscv/zc-zca-c-not-fail-march.d: Likewise.
            * gas/riscv/zc-zca-c-not-fail-march.l: Likewise.
            * gas/riscv/zc-zca-c-not-fail-march.s: Likewise.
            * gas/riscv/zc-zca-cm-lbu-fail-march.d: Likewise.
            * gas/riscv/zc-zca-cm-lbu-fail-march.l: Likewise.
            * gas/riscv/zc-zca-cm-lbu-fail-march.s: Likewise.
            * gas/riscv/zc-zca-cm-lh-fail-march.d: Likewise.
            * gas/riscv/zc-zca-cm-lh-fail-march.l: Likewise.
            * gas/riscv/zc-zca-cm-lh-fail-march.s: Likewise.
            * gas/riscv/zc-zca-cm-lhu-fail-march.d: Likewise.
            * gas/riscv/zc-zca-cm-lhu-fail-march.l: Likewise.
            * gas/riscv/zc-zca-cm-lhu-fail-march.s: Likewise.
            * gas/riscv/zc-zca-cm-mva01s-e-fail-march.d: Likewise.
            * gas/riscv/zc-zca-cm-mva01s-e-fail-march.l: Likewise.
            * gas/riscv/zc-zca-cm-mva01s-e-fail-march.s: Likewise.
            * gas/riscv/zc-zca-cm-mva01s-fail-march.d: Likewise.
            * gas/riscv/zc-zca-cm-mva01s-fail-march.l: Likewise.
            * gas/riscv/zc-zca-cm-mva01s-fail-march.s: Likewise.
            * gas/riscv/zc-zca-cm-mvsa01-e-fail-march.d: Likewise.
            * gas/riscv/zc-zca-cm-mvsa01-e-fail-march.l: Likewise.
            * gas/riscv/zc-zca-cm-mvsa01-e-fail-march.s: Likewise.
            * gas/riscv/zc-zca-cm-mvsa01-fail-march.d: Likewise.
            * gas/riscv/zc-zca-cm-mvsa01-fail-march.l: Likewise.
            * gas/riscv/zc-zca-cm-mvsa01-fail-march.s: Likewise.
            * gas/riscv/zc-zca-cm-pop-e-fail-march.d: Likewise.
            * gas/riscv/zc-zca-cm-pop-e-fail-march.l: Likewise.
            * gas/riscv/zc-zca-cm-pop-e-fail-march.s: Likewise.
            * gas/riscv/zc-zca-cm-pop-fail-march.d: Likewise.
            * gas/riscv/zc-zca-cm-pop-fail-march.l: Likewise.
            * gas/riscv/zc-zca-cm-pop-fail-march.s: Likewise.
            * gas/riscv/zc-zca-cm-popret-e-fail-march.d: Likewise.
            * gas/riscv/zc-zca-cm-popret-e-fail-march.l: Likewise.
            * gas/riscv/zc-zca-cm-popret-e-fail-march.s: Likewise.
            * gas/riscv/zc-zca-cm-popret-fail-march.d: Likewise.
            * gas/riscv/zc-zca-cm-popret-fail-march.l: Likewise.
            * gas/riscv/zc-zca-cm-popret-fail-march.s: Likewise.
            * gas/riscv/zc-zca-cm-popretz-e-fail-march.d: Likewise.
            * gas/riscv/zc-zca-cm-popretz-e-fail-march.l: Likewise.
            * gas/riscv/zc-zca-cm-popretz-e-fail-march.s: Likewise.
            * gas/riscv/zc-zca-cm-popretz-fail-march.d: Likewise.
            * gas/riscv/zc-zca-cm-popretz-fail-march.l: Likewise.
            * gas/riscv/zc-zca-cm-popretz-fail-march.s: Likewise.
            * gas/riscv/zc-zca-cm-push-e-fail-march.d: Likewise.
            * gas/riscv/zc-zca-cm-push-e-fail-march.l: Likewise.
            * gas/riscv/zc-zca-cm-push-e-fail-march.s: Likewise.
            * gas/riscv/zc-zca-cm-push-fail-march.d: Likewise.
            * gas/riscv/zc-zca-cm-push-fail-march.l: Likewise.
            * gas/riscv/zc-zca-cm-push-fail-march.s: Likewise.
            * gas/riscv/zc-zca-cm-sb-fail-march.d: Likewise.
            * gas/riscv/zc-zca-cm-sb-fail-march.l: Likewise.
            * gas/riscv/zc-zca-cm-sb-fail-march.s: Likewise.
            * gas/riscv/zc-zca-cm-sh-fail-march.d: Likewise.
            * gas/riscv/zc-zca-cm-sh-fail-march.l: Likewise.
            * gas/riscv/zc-zca-cm-sh-fail-march.s: Likewise.
            * gas/riscv/zc-zca-jalt-fail-march.d: Likewise.
            * gas/riscv/zc-zca-jalt-fail-march.l: Likewise.
            * gas/riscv/zc-zca-jalt-fail-march.s: Likewise.
            * gas/riscv/zc-zca-jt-fail-march.d: Likewise.
            * gas/riscv/zc-zca-jt-fail-march.l: Likewise.
            * gas/riscv/zc-zca-jt-fail-march.s: Likewise.
            * gas/riscv/zc-zca-lb-fail-march.d: Likewise.
            * gas/riscv/zc-zca-lb-fail-march.l: Likewise.
            * gas/riscv/zc-zca-lb-fail-march.s: Likewise.
            * gas/riscv/zc-zca-lbu-fail-march.d: Likewise.
            * gas/riscv/zc-zca-lbu-fail-march.l: Likewise.
            * gas/riscv/zc-zca-lbu-fail-march.s: Likewise.
            * gas/riscv/zc-zca-lh-fail-march.d: Likewise.
            * gas/riscv/zc-zca-lh-fail-march.l: Likewise.
            * gas/riscv/zc-zca-lh-fail-march.s: Likewise.
            * gas/riscv/zc-zca-lhu-fail-march.d: Likewise.
            * gas/riscv/zc-zca-lhu-fail-march.l: Likewise.
            * gas/riscv/zc-zca-lhu-fail-march.s: Likewise.
            * gas/riscv/zc-zca-march-accept.s: Likewise.
            * gas/riscv/zc-zca-sb-fail-march.d: Likewise.
            * gas/riscv/zc-zca-sb-fail-march.l: Likewise.
            * gas/riscv/zc-zca-sb-fail-march.s: Likewise.
            * gas/riscv/zc-zca-sext-b-fail-march.d: Likewise.
            * gas/riscv/zc-zca-sext-b-fail-march.l: Likewise.
            * gas/riscv/zc-zca-sext-b-fail-march.s: Likewise.
            * gas/riscv/zc-zca-sext-h-fail-march.d: Likewise.
            * gas/riscv/zc-zca-sext-h-fail-march.l: Likewise.
            * gas/riscv/zc-zca-sext-h-fail-march.s: Likewise.
            * gas/riscv/zc-zca-sh-fail-march.d: Likewise.
            * gas/riscv/zc-zca-sh-fail-march.l: Likewise.
            * gas/riscv/zc-zca-sh-fail-march.s: Likewise.
            * gas/riscv/zc-zca-zext-b-fail-march.d: Likewise.
            * gas/riscv/zc-zca-zext-b-fail-march.l: Likewise.
            * gas/riscv/zc-zca-zext-b-fail-march.s: Likewise.
            * gas/riscv/zc-zca-zext-h-fail-march.d: Likewise.
            * gas/riscv/zc-zca-zext-h-fail-march.l: Likewise.
            * gas/riscv/zc-zca-zext-h-fail-march.s: Likewise.
            * gas/riscv/zc-zca-zext-w-fail-march.d: Likewise.
            * gas/riscv/zc-zca-zext-w-fail-march.l: Likewise.
            * gas/riscv/zc-zca-zext-w-fail-march.s: Likewise.

Commit:
Added test cases for invalid opcodes on Zca

	* gas/riscv/zc-zca-c-flw-c-flwsp-c-fsw-c-fswsp-fail-march.d: Added test case for failure on use of invalid opcodes for Zca.
	* gas/riscv/zc-zca-c-flw-c-flwsp-c-fsw-c-fswsp-fail-march.l: Likewise.
	* gas/riscv/zc-zca-c-flw-c-flwsp-c-fsw-c-fswsp-fail-march.s: Likewise.
	* gas/riscv/zc-zca-c-mul-fail-march.d: Likewise.
	* gas/riscv/zc-zca-c-mul-fail-march.l: Likewise.
	* gas/riscv/zc-zca-c-mul-fail-march.s: Likewise.
	* gas/riscv/zc-zca-c-not-fail-march.d: Likewise.
	* gas/riscv/zc-zca-c-not-fail-march.l: Likewise.
	* gas/riscv/zc-zca-c-not-fail-march.s: Likewise.
	* gas/riscv/zc-zca-cm-lbu-fail-march.d: Likewise.
	* gas/riscv/zc-zca-cm-lbu-fail-march.l: Likewise.
	* gas/riscv/zc-zca-cm-lbu-fail-march.s: Likewise.
	* gas/riscv/zc-zca-cm-lh-fail-march.d: Likewise.
	* gas/riscv/zc-zca-cm-lh-fail-march.l: Likewise.
	* gas/riscv/zc-zca-cm-lh-fail-march.s: Likewise.
	* gas/riscv/zc-zca-cm-lhu-fail-march.d: Likewise.
	* gas/riscv/zc-zca-cm-lhu-fail-march.l: Likewise.
	* gas/riscv/zc-zca-cm-lhu-fail-march.s: Likewise.
	* gas/riscv/zc-zca-cm-mva01s-e-fail-march.d: Likewise.
	* gas/riscv/zc-zca-cm-mva01s-e-fail-march.l: Likewise.
	* gas/riscv/zc-zca-cm-mva01s-e-fail-march.s: Likewise.
	* gas/riscv/zc-zca-cm-mva01s-fail-march.d: Likewise.
	* gas/riscv/zc-zca-cm-mva01s-fail-march.l: Likewise.
	* gas/riscv/zc-zca-cm-mva01s-fail-march.s: Likewise.
	* gas/riscv/zc-zca-cm-mvsa01-e-fail-march.d: Likewise.
	* gas/riscv/zc-zca-cm-mvsa01-e-fail-march.l: Likewise.
	* gas/riscv/zc-zca-cm-mvsa01-e-fail-march.s: Likewise.
	* gas/riscv/zc-zca-cm-mvsa01-fail-march.d: Likewise.
	* gas/riscv/zc-zca-cm-mvsa01-fail-march.l: Likewise.
	* gas/riscv/zc-zca-cm-mvsa01-fail-march.s: Likewise.
	* gas/riscv/zc-zca-cm-pop-e-fail-march.d: Likewise.
	* gas/riscv/zc-zca-cm-pop-e-fail-march.l: Likewise.
	* gas/riscv/zc-zca-cm-pop-e-fail-march.s: Likewise.
	* gas/riscv/zc-zca-cm-pop-fail-march.d: Likewise.
	* gas/riscv/zc-zca-cm-pop-fail-march.l: Likewise.
	* gas/riscv/zc-zca-cm-pop-fail-march.s: Likewise.
	* gas/riscv/zc-zca-cm-popret-e-fail-march.d: Likewise.
	* gas/riscv/zc-zca-cm-popret-e-fail-march.l: Likewise.
	* gas/riscv/zc-zca-cm-popret-e-fail-march.s: Likewise.
	* gas/riscv/zc-zca-cm-popret-fail-march.d: Likewise.
	* gas/riscv/zc-zca-cm-popret-fail-march.l: Likewise.
	* gas/riscv/zc-zca-cm-popret-fail-march.s: Likewise.
	* gas/riscv/zc-zca-cm-popretz-e-fail-march.d: Likewise.
	* gas/riscv/zc-zca-cm-popretz-e-fail-march.l: Likewise.
	* gas/riscv/zc-zca-cm-popretz-e-fail-march.s: Likewise.
	* gas/riscv/zc-zca-cm-popretz-fail-march.d: Likewise.
	* gas/riscv/zc-zca-cm-popretz-fail-march.l: Likewise.
	* gas/riscv/zc-zca-cm-popretz-fail-march.s: Likewise.
	* gas/riscv/zc-zca-cm-push-e-fail-march.d: Likewise.
	* gas/riscv/zc-zca-cm-push-e-fail-march.l: Likewise.
	* gas/riscv/zc-zca-cm-push-e-fail-march.s: Likewise.
	* gas/riscv/zc-zca-cm-push-fail-march.d: Likewise.
	* gas/riscv/zc-zca-cm-push-fail-march.l: Likewise.
	* gas/riscv/zc-zca-cm-push-fail-march.s: Likewise.
	* gas/riscv/zc-zca-cm-sb-fail-march.d: Likewise.
	* gas/riscv/zc-zca-cm-sb-fail-march.l: Likewise.
	* gas/riscv/zc-zca-cm-sb-fail-march.s: Likewise.
	* gas/riscv/zc-zca-cm-sh-fail-march.d: Likewise.
	* gas/riscv/zc-zca-cm-sh-fail-march.l: Likewise.
	* gas/riscv/zc-zca-cm-sh-fail-march.s: Likewise.
	* gas/riscv/zc-zca-jalt-fail-march.d: Likewise.
	* gas/riscv/zc-zca-jalt-fail-march.l: Likewise.
	* gas/riscv/zc-zca-jalt-fail-march.s: Likewise.
	* gas/riscv/zc-zca-jt-fail-march.d: Likewise.
	* gas/riscv/zc-zca-jt-fail-march.l: Likewise.
	* gas/riscv/zc-zca-jt-fail-march.s: Likewise.
	* gas/riscv/zc-zca-lb-fail-march.d: Likewise.
	* gas/riscv/zc-zca-lb-fail-march.l: Likewise.
	* gas/riscv/zc-zca-lb-fail-march.s: Likewise.
	* gas/riscv/zc-zca-lbu-fail-march.d: Likewise.
	* gas/riscv/zc-zca-lbu-fail-march.l: Likewise.
	* gas/riscv/zc-zca-lbu-fail-march.s: Likewise.
	* gas/riscv/zc-zca-lh-fail-march.d: Likewise.
	* gas/riscv/zc-zca-lh-fail-march.l: Likewise.
	* gas/riscv/zc-zca-lh-fail-march.s: Likewise.
	* gas/riscv/zc-zca-lhu-fail-march.d: Likewise.
	* gas/riscv/zc-zca-lhu-fail-march.l: Likewise.
	* gas/riscv/zc-zca-lhu-fail-march.s: Likewise.
	* gas/riscv/zc-zca-march-accept.s: Likewise.
	* gas/riscv/zc-zca-sb-fail-march.d: Likewise.
	* gas/riscv/zc-zca-sb-fail-march.l: Likewise.
	* gas/riscv/zc-zca-sb-fail-march.s: Likewise.
	* gas/riscv/zc-zca-sext-b-fail-march.d: Likewise.
	* gas/riscv/zc-zca-sext-b-fail-march.l: Likewise.
	* gas/riscv/zc-zca-sext-b-fail-march.s: Likewise.
	* gas/riscv/zc-zca-sext-h-fail-march.d: Likewise.
	* gas/riscv/zc-zca-sext-h-fail-march.l: Likewise.
	* gas/riscv/zc-zca-sext-h-fail-march.s: Likewise.
	* gas/riscv/zc-zca-sh-fail-march.d: Likewise.
	* gas/riscv/zc-zca-sh-fail-march.l: Likewise.
	* gas/riscv/zc-zca-sh-fail-march.s: Likewise.
	* gas/riscv/zc-zca-zext-b-fail-march.d: Likewise.
	* gas/riscv/zc-zca-zext-b-fail-march.l: Likewise.
	* gas/riscv/zc-zca-zext-b-fail-march.s: Likewise.
	* gas/riscv/zc-zca-zext-h-fail-march.d: Likewise.
	* gas/riscv/zc-zca-zext-h-fail-march.l: Likewise.
	* gas/riscv/zc-zca-zext-h-fail-march.s: Likewise.
	* gas/riscv/zc-zca-zext-w-fail-march.d: Likewise.
	* gas/riscv/zc-zca-zext-w-fail-march.l: Likewise.
	* gas/riscv/zc-zca-zext-w-fail-march.s: Likewise.

Commit:
Added test case for failure or success on use of invalid or valid opcodes respectively for Zcb.

Commit:
Added test cases for success on use of valid opcodes for Zcf.

    * gas/riscv/zc-zcf-march-accept.d: Added test case for success on use of valid opcodes for Zcf.
	* gas/riscv/zc-zcf-march-accept.s: Added test case for success on use of valid opcodes for Zcf.
	* gas/riscv/zc-zcf-march-reject.d: Added test case for failure on use of invalid opcodes for Zcf.
	* gas/riscv/zc-zcf-march-reject.l: Added test case for failure on use of invalid opcodes for Zcf.
	* gas/riscv/zc-zcf-march-reject.s: Added test case for failure on use of invalid opcodes for Zcf.

Commit:
Added test case for failure/success on use of invalid/valid opcodes respectively for Zcmb.

	* gas/riscv/zc-zcmb-march-accept.d: Added test case for success on use of valid opcodes for Zcmb.
	* gas/riscv/zc-zcmb-march-accept.s: Added test case for success on use of valid opcodes for Zcmb.
	* gas/riscv/zc-zcmb-march-reject.d: Added test case for failure on use of invalid opcodes for Zcmb.
	* gas/riscv/zc-zcmb-march-reject.l: Added test case for failure on use of invalid opcodes for Zcmb.
	* gas/riscv/zc-zcmb-march-reject.s: Added test case for failure on use of invalid opcodes for Zcmb.

Commit:
Added test case for failure/success on use of valid/invalid opcodes for Zcmp respectively.

	* gas/riscv/zc-zcmp-march-accept.d: Added test case for success on use of valid opcodes for Zcmp.
	* gas/riscv/zc-zcmp-march-accept.s: Added test case for success on use of valid opcodes for Zcmp.
	* gas/riscv/zc-zcmp-march-reject.d: Added test case for failure on use of invalid opcodes for Zcmp.
	* gas/riscv/zc-zcmp-march-reject.l: Added test case for failure on use of invalid opcodes for Zcmp.
	* gas/riscv/zc-zcmp-march-reject.s: Added test case for failure on use of invalid opcodes for Zcmp.

Commit:
Added test case for failure/success on use of invalid/valid opcodes respectively for Zcmpe.

	* gas/riscv/zc-zcmpe-march-accept.d: Added test case for success on use of valid opcodes for Zcmpe.
	* gas/riscv/zc-zcmpe-march-accept.s: Added test case for success on use of valid opcodes for Zcmpe.
	* gas/riscv/zc-zcmpe-march-reject.d: Added test case for failure on use of invalid opcodes for Zcmpe.
	* gas/riscv/zc-zcmpe-march-reject.l: Added test case for failure on use of invalid opcodes for Zcmpe.
	* gas/riscv/zc-zcmpe-march-reject.s: Added test case for failure on use of invalid opcodes for Zcmpe.

Commit:
Added test case for failure/success on use of invalid/valid opcodes respectively for Zcmt.

    * gas/testsuite/gas/riscv/zc-zcmt-march-accept.d: Added test case for success on use of valid opcodes for Zcmt.
	* gas/testsuite/gas/riscv/zc-zcmt-march-accept.s: Added test case for success on use of valid opcodes for Zcmt.
	* gas/testsuite/gas/riscv/zc-zcmt-march-reject.d: Added test case for failure on use of invalid opcodes for Zcmt.
	* gas/testsuite/gas/riscv/zc-zcmt-march-reject.l: Added test case for failure on use of invalid opcodes for Zcmt.
	* gas/testsuite/gas/riscv/zc-zcmt-march-reject.s: Added test case for failure on use of invalid opcodes for Zcmt.

Commit:
Fixed bug that permitted use of invalid operands for cm.jt

	* gas/config/tc-riscv.c: Fixed bug that allowed use of invalid operand 64 with cm.jt.

Commit:
Added tests for failure on invalid immediate offset outside of range for Zc* extensions.

	* gas/riscv/zc-zcb-c-lbu-range-fail.d: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-c-lbu-range-fail.l: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-c-lbu-range-fail.s: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-c-lh-range-fail.d: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-c-lh-range-fail.l: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-c-lh-range-fail.s: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-c-lhu-range-fail.d: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-c-lhu-range-fail.l: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-c-lhu-range-fail.s: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-c-sb-range-fail.d: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-c-sb-range-fail.l: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-c-sb-range-fail.s: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-c-sh-range-fail.d: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-c-sh-range-fail.l: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-c-sh-range-fail.s: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-cm-lh-range-fail.d: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-cm-lh-range-fail.l: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-cm-lh-range-fail.s: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-cm-lhu-range-fail.d: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-cm-lhu-range-fail.l: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-cm-lhu-range-fail.s: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcb-cm-lu-range-fail.l: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcmt-jalt-range-fail.d: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcmt-jalt-range-fail.l: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcmt-jalt-range-fail.s: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcmt-jt-range-fail.d: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcmt-jt-range-fail.l: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcmt-jt-range-fail.s: Added test for failure on invalid immediate offset outside of range.
	* gas/riscv/zc-zcmt-march-accept.d: Added test case for success on use of valid opcodes for Zcmt.
	* gas/riscv/zc-zcmt-march-accept.s: Added test case for success on use of valid opcodes for Zcmt.
	* gas/riscv/zc-zcmt-march-reject.d: Added test case for failure on use of invalid opcodes for Zcmt.
	* gas/riscv/zc-zcmt-march-reject.l: Added test case for failure on use of invalid opcodes for Zcmt.
	* gas/riscv/zc-zcmt-march-reject.s: Added test case for failure on use of invalid opcodes for Zcmt.

Commit:
Added test for successful assembly of cm.jt and cm.jalt instructions.

	* gas/riscv/zc-zcmt-jalt.d: Added test for successful assembly of instruction.
	* gas/riscv/zc-zcmt-jalt.s: Added test for successful assembly of instruction.
	* gas/riscv/zc-zcmt-jt.d: Added test for successful assembly of instruction.
	* gas/riscv/zc-zcmt-jt.s: Added test for successful assembly of instruction.

Commit:
Fixed typo in zcmb/zcmpe error message.

    * bfd/elfxx-riscv.c: Fixed typo in error message.
	* gas/testsuite/gas/riscv/zc-zcmb-fail-arch-conflict-0.l: Fixed typo in error message.
	* gas/testsuite/gas/riscv/zc-zcmb-fail-arch-conflict-1.l: Fixed typo in error message.
	* gas/testsuite/gas/riscv/zc-zcmpe-fail-arch-conflict-1.l: Fixed typo in error message.

Commit:
Made Zca and C being equivalent instruction classes explicit.

    * include/opcode/riscv.h: Made Zca and C being equivalent instruction classes explicit.
    * opcodes/riscv-opc.c: Made Zca and C being equivalent instruction classes explicit.

simonpcook commented on bfd/ChangeLog:
```diff
@@ -1,3 +1,7 @@
+2022-01-29  Charlie Keaney  <charlie.keaney@embecosm.com>
+
+	* bfd/elfxx-riscv.c: Fixed typo in error message.
+
```
I would probably keep this in a `ChangeLog.COREV` to keep in sync with how e.g. `gas/testsuite/ChangeLog.COREV` would exist.

simonpcook commented on gas/ChangeLog:
```diff
@@ -1,3 +1,6 @@
+2022-07-29  Charlie Keaney  <charlie.keaney@embecosm.com>
+	* gas/config/tc-riscv.c: Fixed bug that allowed use of invalid operand 64 with cm.jt.
```
Missing blank line, same point about `ChangeLog.COREV`

CharKeaney replied:
Fixed in next patch.

simonpcook commented on gas/testsuite/ChangeLog.COREV:
```diff
@@ -1,3 +1,188 @@
+	* gas/riscv/zc-zcmt-jalt.s: Added test for successful assembly of instruction.
+	* gas/riscv/zc-zcmt-jt.d: Added test for successful assembly of instruction.
+	* gas/riscv/zc-zcmt-jt.s: Added test for successful assembly of instruction.
+	* gas/riscv/zc-zcb-c-lbu-range-fail.d: Added test for failure on invalid immediate offset outside of range.
```
These should be flowed to 80 columns.

Also for consistency, I see some `Likewise`s, I would use these throughout if it is the same thing.

CharKeaney replied:
Fixed in next patch.

simonpcook commented on gas/testsuite/ChangeLog.COREV:
```diff
@@ -1,3 +1,188 @@
+2022-07-29  Charlie Keaney  <charlie.keaney@embecosm.com>
+
```
These two lines should be added for each patch in which you're adding a new entry.

simonpcook commented on gas/testsuite/gas/riscv/zc-zca-c-mul-fail-march.d:
```diff
@@ -0,0 +1,3 @@
+#as: -march=rv32ifd_zca
+#source: zc-zca-c-mul-fail-march.s
+#error_output: zc-zca-c-mul-fail-march.l
```
No newline at end of file.

CharKeaney replied:
Fixed in next patch.

simonpcook commented on gas/testsuite/gas/riscv/zc-zca-cm-lbu-fail-march.d:
```diff
@@ -0,0 +1,3 @@
+#as: -march=rv64ifd_zca
+#source: zc-zca-cm-lbu-fail-march.s
+#error_output: zc-zca-cm-lbu-fail-march.l
```
No newline at end of file.

CharKeaney replied:
Fixed in next patch.

simonpcook commented on gas/testsuite/gas/riscv/zc-zca-c-not-fail-march.s:
```diff
@@ -0,0 +1,9 @@
+	# test c.not
+	c.not x9
+	c.not x11
+    c.not x12
```
This instruction is misaligned.

CharKeaney replied:
Fixed in next patch.

simonpcook commented on gas/testsuite/gas/riscv/zc-zca-cm-lh-fail-march.d:
```diff
@@ -0,0 +1,3 @@
+#as: -march=rv64ifd_zca
+#source: zc-zca-cm-lh-fail-march.s
+#error_output: zc-zca-cm-lh-fail-march.l
```
No newline at end of file.

(I'll stop commenting on these down, but check all these)

CharKeaney replied:
Fixed in next patch.

simonpcook commented on gas/testsuite/gas/riscv/zc-zca-cm-lh-fail-march.s:
```diff
@@ -0,0 +1,9 @@
+zcb_lh:
+	# test to compress lh
```
nit: comments like this don't add much, from the file name and the symbol name that's implied.

CharKeaney replied:
Fixed in next patch.

simonpcook commented on gas/testsuite/gas/riscv/zc-zca-cm-popretz-e-fail-march.s:
```diff
@@ -0,0 +1,45 @@
+    cm.popretz.e {ra, s0-s8}, 64
+    cm.popretz.e {ra, s0-s11}, 64
+
+    # numeric .enames
```
I don't understand what you mean by this?

CharKeaney replied:
Fixed in next patch.

simonpcook commented on gas/testsuite/gas/riscv/zc-zcb-cm-lh-range-fail.s:
```diff
@@ -0,0 +1,20 @@
+target:
+	cm.lh x8,-1(x8)
+	cm.lh x8,1(x8)
+    cm.lh x8,3(x8)
```
misalignment

CharKeaney replied:
Fixed in next patch.

simonpcook commented on gas/testsuite/gas/riscv/zc-zcb-cm-lhu-range-fail.s:
```diff
@@ -0,0 +1,20 @@
+target:
+	cm.lhu x8,-1(x8)
+	cm.lhu x8,1(x8)
+    cm.lhu x8,3(x8)
```
misalignment

CharKeaney replied:
Fixed in next patch.

simonpcook commented on gas/testsuite/gas/riscv/zc-zcb-m-march-reject.s:
```diff
@@ -0,0 +1,30 @@
+    c.zext.w x8
+    cm.lb x8,(x8)
+    cm.lbu x8,(x8)
+	cm.lh x8,2(x8)
+	cm.lhu x8,2(x8)
```
Alignment

CharKeaney replied:
Fixed in next patch.

simonpcook commented on gas/testsuite/ChangeLog.COREV:
```diff
@@ -1,3 +1,188 @@
+	* gas/riscv/zc-zcb-march-reject.d: Added test case for failure on use of invalid opcodes for Zcb.
+	* gas/riscv/zc-zcb-march-reject.l: Added test case for failure on use of invalid opcodes for Zcb.
+	* gas/riscv/zc-zcb-march-reject.s: Added test case for failure on use of invalid opcodes for Zcb.
+	* gas/riscv/zc-zcb-mul.s: Fixed typo.
+	* gas/riscv/zc-zcb-not.s: Fixed typo.
+	* gas/riscv/zc-zcb-sext-b.s: Fixed typo.
+	* gas/riscv/zc-zcb-sext-h.s: Fixed Typo.
```
I don't think I saw these in the commit message?

CharKeaney replied:
> I've done a brief pass-through, though haven't checked correctness (I'm leaving that to @pietraferreira), but a couple of other things I couldn't comment on:
> 
>     * `gas/testsuite/gas/riscv/zc-zca-march-accept.s`: Why not use `empty.s`?
> 
>     * `gas/testsuite/gas/riscv/zc-zcb-cm-lu-range-fail.l`: Why is this file empty?

Fixed both those issues in next patch

simonpcook commented on gas/testsuite/gas/riscv/zc-zcb-zext-b.s:
```diff
@@ -1,12 +1,12 @@
 	c.zext.b x12
 
-	# test to compress sign/zero extension insturctions
+	# test to compress sign/zero extension instructions 
```
nit: whitespace at EOL

CharKeaney replied:
Fixed in next patch.

simonpcook commented on gas/testsuite/gas/riscv/zc-zcmt-march-reject.s:
```diff
@@ -0,0 +1,36 @@
+    cm.lb x8,(x8)
+    cm.lbu x8,(x8)
+    cm.lh x8,2(x8)
+	cm.lhu x8,2(x8)
```
alignment

CharKeaney replied:
Fixed in next patch.

simonpcook commented on include/ChangeLog:
```diff
@@ -1,3 +1,7 @@
+2022-07-29  Charlie Keaney  <charlie.keaney@embecosm.com>
+
+	opcode/riscv.h: Made Zca and C being equivalent instruction classes explicit.
+
```
`ChangeLog.COREV`

simonpcook commented:
I've done a brief pass-through, though haven't checked correctness (I'm leaving that to @pietraferreira), but a couple of other things I couldn't comment on:

* `gas/testsuite/gas/riscv/zc-zca-march-accept.s`: Why not use `empty.s`?
* `gas/testsuite/gas/riscv/zc-zcb-cm-lu-range-fail.l`: Why is this file empty?

Commit:
Deleted unused test files.

	* gas/testsuite/gas/riscv/zc-zca-march-accept.s: Deleted unused file.
	* gas/testsuite/gas/riscv/zc-zcb-cm-lu-range-fail.l: Deleted unused file.

Commit:
Fix whitespace, changelog issues, and remove unnecessary comments

	* gas/testsuite/gas/riscv/zc-zca-c-mul-fail-march.d: Fixed
	 whitespace.
	* gas/testsuite/gas/riscv/zc-zca-c-not-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-lbu-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-lh-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-lh-fail-march.s: Removed
	unnecessary comment.
	* gas/testsuite/gas/riscv/zc-zca-cm-lhu-fail-march.d: Fixed
	whitespace.
	* gas/testsuite/gas/riscv/zc-zca-cm-mva01s-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-mvsa01-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-pop-e-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-pop-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-popret-e-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-popret-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-popretz-e-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-popretz-e-fail-march.s: Removed
	unnecessary comment.
	* gas/testsuite/gas/riscv/zc-zca-cm-popretz-fail-march.d: Fixed
	whitespace.
	* gas/testsuite/gas/riscv/zc-zca-cm-push-e-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-push-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-sb-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-sh-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-ebreak-rv32-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-ebreak-rv32-fail.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-ebreak-rv64-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-ebreak-rv64-fail.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-j-rv32-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-j-rv32-fail.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-j-rv32.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-j-rv64-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-j-rv64.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-jalr-rv32-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-jalr-rv32-fail.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-jalr-rv64-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-jalr-rv64-fail.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-jalt-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-jr-rv32-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-jr-rv32-fail.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-jr-rv64-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-jr-rv64-fail.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-jt-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lb-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lbu-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lh-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lhu-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-li-rv32-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-li-rv32.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-li-rv64-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lui-rv32-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lui-rv64-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lw-rv32-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lw-rv64-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lw-rv64-fail.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lwsp-rv32-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lwsp-rv32.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lwsp-rv64-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-mv-rv32-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-mv-rv32-fail.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-mv-rv64-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-nop-rv32-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-nop-rv64-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-rv32-jal-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-rv32-jal-fail.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-rv32-jal.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-sb-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-sext-b-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-sext-h-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-sh-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-sw-rv32-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zcb-c-lbu-range-fail.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zcb-cm-lh-range-fail.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zcb-cm-lhu-range-fail.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zcb-m-march-reject.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zcb-zext-b.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zcmp-pop-rv64-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zcmp-popret-rv32-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zcmp-popret-rv64-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zcmp-popretz-rv32-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zcmp-popretz-rv64-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zcmp-push-rv32-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zcmp-push-rv64-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zcmpe-pop-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zcmpe-popret-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zcmpe-popretz-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zcmpe-push-fail.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zcmt-march-reject.s: Likewise.
	* bfd/ChangeLog: Deleted most recent entry.
	* bfd/ChangeLog.COREV: Added new entry.
	* gas/testsuite/ChangeLog.COREV: Fixed spacing.

Commit:
Removed unnecessary comment.

	* gas/testsuite/ChangeLog.COREV
	* gas/testsuite/gas/riscv/zc-zca-cm-lh-fail-march.s

Commit:
Moved zca fail march tests into a single, new, test case.

	* gas/testsuite/gas/riscv/zc-zca-c-flw-c-flwsp-c-fsw-c-fswsp-fail-march.d:
	Deleted and moved its test into zc-zca-march-reject.
	* gas/testsuite/gas/riscv/zc-zca-c-flw-c-flwsp-c-fsw-c-fswsp-fail-march.l:
	Likewise.
	* gas/testsuite/gas/riscv/zc-zca-c-flw-c-flwsp-c-fsw-c-fswsp-fail-march.s:
	Likewise.
	* gas/testsuite/gas/riscv/zc-zca-c-mul-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-c-mul-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-c-mul-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-c-not-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-c-not-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-c-not-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lbu-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lbu-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lbu-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lhu-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lhu-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lhu-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-march-reject.d: Added new test case.
	* gas/testsuite/gas/riscv/zc-zca-march-reject.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-march-reject.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-sext-b-fail-march.d: Deleted and moved its
	test into zc-zca-march-reject.
	* gas/testsuite/gas/riscv/zc-zca-sext-b-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-sext-b-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-sext-h-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-sext-h-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-sext-h-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-sh-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-sh-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-sh-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-zext-b-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-zext-b-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-zext-b-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-zext-h-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-zext-h-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-zext-h-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-zext-w-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-zext-w-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-zext-w-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-lbu-fail-march.d: Moved test into
	zc-zca-march-reject and deleted original.
	* gas/testsuite/gas/riscv/zc-zca-cm-lbu-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-lbu-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-lh-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-lh-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-lh-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-lhu-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-lhu-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-lhu-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-mva01s-e-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-mva01s-e-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-mva01s-e-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-mva01s-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-mva01s-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-mva01s-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-mvsa01-e-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-mvsa01-e-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-mvsa01-e-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-mvsa01-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-mvsa01-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-mvsa01-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-pop-e-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-pop-e-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-pop-e-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-pop-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-pop-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-pop-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-popret-e-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-popret-e-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-popret-e-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-popret-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-popret-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-popret-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-popretz-e-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-popretz-e-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-popretz-e-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-popretz-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-popretz-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-popretz-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-push-e-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-push-e-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-push-e-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-push-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-push-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-push-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-sb-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-sb-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-sb-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-sh-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-sh-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-cm-sh-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-jalt-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-jalt-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-jalt-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-jt-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-jt-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-jt-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lb-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lb-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lb-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lh-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lh-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-lh-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-march-reject.l: Expanded test case.
	* gas/testsuite/gas/riscv/zc-zca-march-reject.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-sb-fail-march.d: Moved test into
	zc-zca-march-reject and deleted original.
	* gas/testsuite/gas/riscv/zc-zca-sb-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zca-sb-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zcb-zba-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zcb-zba-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zcb-zba-fail-march.s: Likewise.
	* gas/testsuite/gas/riscv/zc-zcb-zbb-fail-march.d: Likewise.
	* gas/testsuite/gas/riscv/zc-zcb-zbb-fail-march.l: Likewise.
	* gas/testsuite/gas/riscv/zc-zcb-zbb-fail-march.s: Likewise.

simonpcook commented:
To check, I see you're pushing new commits on top, is the idea you're going to make this fixups for the initial patch series?
